rust/basic_ethereum/src/lib.rs

```diff
@@ -52,18 +56,24 @@ struct JsonRpcError {
 #[update]
 pub async fn get_balance(address: String) -> Nat {
     let _caller = validate_caller_not_anonymous();
-    let chain_id = read_state(|s| s.ethereum_network().chain_id());
     let json = format!(
-        r#"{{ "jsonrpc": "2.0", "method": "eth_getBalance", "params": ["{}", "latest"], "id": {} }}"#,
-        address, chain_id
+        r#"{{ "jsonrpc": "2.0", "method": "eth_getBalance", "params": ["{}", "latest"], "id": 1 }}"#,
+        address
     );
+
+    let max_response_size_bytes = 500_u64;
+    let num_cycles = 1_000_000_000u128;
+
+    let chain_id = read_state(|s| s.ethereum_network().chain_id());
+
+    let rpc_service = match chain_id {
+        ETH_MAINNET_CHAIN_ID => RpcService::EthMainnet(EthMainnetService::PublicNode),
+        ETH_SEPOLIA_CHAIN_ID => RpcService::EthSepolia(EthSepoliaService::PublicNode),
+        _ => panic!("Unsupported chain ID in get_balance call."),
+    };
```

Comment:
note that the 3rd case cannot happen, this is an artifact of calling `chain_id()`. This should work better

```suggestion
    let ethereum_network = read_state(|s| s.ethereum_network());

    let rpc_service = match ethereum_network {
        EthereumNetwork::Mainnet => RpcService::EthMainnet(EthMainnetService::PublicNode),
        EthereumNetwork::Sepolia => RpcService::EthSepolia(EthSepoliaService::PublicNode),
    };
```